### ISSUE TITLE
NGINX problem

### ISSUE THREAD

**Issue by szilardshomelab** [pabloromeo/clusterplex#323]
I created a deployment for k3s cluster with remote transcoders.

After PMS start I receive the following error :

nginx: [emerg] socket() [::]:32499 failed (97: Unknown error)

I need to manually edit the NGINX forward-proxy.conf from this:

 listen [::]:32499 ipv6only=off;

To this:

 listen 32499 ipv6only=off;

But unfortunately if I recreate the pod the modification will gone.

Here is my deployment.yaml code:

```
kind: ConfigMap
apiVersion: v1
metadata:
  name: clusterplex-pms-config
  labels:
    app.kubernetes.io/name: clusterplex-pms-config
    app.kubernetes.io/part-of: clusterplex
  namespace: plex
data:
  VERSION: docker
  TZ: Europe/Bucharest
  PGID: '1000'
  PUID: '1000'
  DOCKER_MODS: 'ghcr.io/pabloromeo/clusterplex_dockermod:latest'
  ORCHESTRATOR_URL: 'http://clusterplex-orchestrator:3500'
  PMS_SERVICE: "clusterplex-pms"
  PMS_PORT: "32400"
  TRANSCODER_VERBOSE: '1'
  TRANSCODE_OPERATING_MODE: "remote"
  LOCAL_RELAY_ENABLED: "1"
  LOCAL_RELAY_PORT: "32499"
  PLEX_CLAIM: "claim-MaQ6dyGzRxUsSPMy9dAh"
---
kind: ConfigMap
apiVersion: v1
metadata:
  name: clusterplex-orchestrator-config
  labels:
    app.kubernetes.io/name: clusterplex-orchestrator-config
    app.kubernetes.io/part-of: clusterplex
  namespace: plex
data:
  TZ: Europe/Bucharest
  LISTENING_PORT: '3500'
  WORKER_SELECTION_STRATEGY: LOAD_TASKS
---
kind: ConfigMap
apiVersion: v1
metadata:
  name: clusterplex-worker-config
  labels:
    app.kubernetes.io/name: clusterplex-worker-config
    app.kubernetes.io/part-of: clusterplex
  namespace: plex
data:
  TZ: Europe/Bucharest
  PGID: '1000'
  PUID: '1000'
  VERSION: docker
  DOCKER_MODS: 'ghcr.io/pabloromeo/clusterplex_worker_dockermod:latest'
  ORCHESTRATOR_URL: 'http://clusterplex-orchestrator:3500'
  LISTENING_PORT: '3501'
  STAT_CPU_INTERVAL: '10000'
  EAE_SUPPORT: '1'
---
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: clusterplex-config-pvc
  labels:
    app.kubernetes.io/name: clusterplex-config-pvc
    app.kubernetes.io/part-of: clusterplex
  namespace: plex
spec:
  accessModes:
  - ReadWriteMany
  resources:
    requests:
      storage: "20Gi"
  # specify your storage class
  storageClassName: longhorn
---
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: clusterplex-transcode-pvc
  labels:
    app.kubernetes.io/name: clusterplex-transcode-pvc
    app.kubernetes.io/part-of: clusterplex
  namespace: plex
spec:
  accessModes:
  - ReadWriteMany
  resources:
    requests:
      storage: "175Gi"
  # specify your storage class
  storageClassName: longhorn
---
apiVersion: v1
kind: PersistentVolume
metadata:
  annotations:
    pv.kubernetes.io/provisioned-by: smb.csi.k8s.io
  name: clusterplex-media
spec:
  capacity:
    storage: 21Ti
  accessModes:
    - ReadWriteMany
  persistentVolumeReclaimPolicy: Retain
  storageClassName: smb
  mountOptions:
    - dir_mode=0777
    - file_mode=0777
  csi:
    driver: smb.csi.k8s.io
    readOnly: false
    volumeHandle: plex
    volumeAttributes:
      source: "//10.10.10.100/data" 
    nodeStageSecretRef:
      name: smbcreds
      namespace: default
---
kind: PersistentVolumeClaim
apiVersion: v1
metadata:
  name: clusterplex-media-pvc
  namespace: plex
spec:
  accessModes:
    - ReadWriteMany
  resources:
    requests:
      storage: 21Ti
  volumeName: clusterplex-media
  storageClassName: smb
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: clusterplex-pms
  labels:
    app.kubernetes.io/name: clusterplex-pms
    app.kubernetes.io/part-of: clusterplex
  namespace: plex
spec:
  replicas: 1
  selector:
    matchLabels:
      app.kubernetes.io/name: clusterplex-pms
      app.kubernetes.io/part-of: clusterplex
  template:
    metadata:
      labels:
        app.kubernetes.io/name: clusterplex-pms
        app.kubernetes.io/part-of: clusterplex
      #annotations:
        # k8s.v1.cni.cncf.io/networks: |
        #  [
        #    {
        #      "name": "multus-untrusted",
        #      "namespace": "default",
        #      "mac": "3a:2e:73:1a:51:9d",
        #      "ips": ["10.10.60.101/24"]
        #    }
        #  ]
    spec:
      affinity:
        podAntiAffinity:
          preferredDuringSchedulingIgnoredDuringExecution:
          - podAffinityTerm:
              labelSelector:
                matchLabels:
                  name: clusterplex-worker
              topologyKey: kubernetes.io/hostname
            weight: 50
          - podAffinityTerm:
              labelSelector:
                matchLabels:
                  name: clusterplex-pms
              topologyKey: kubernetes.io/hostname
            weight: 100
      containers:
      - name: plex
        image: ghcr.io/pabloromeo/clusterplex_pms:latest
        ports:
          - name: pms
            containerPort: 32400
          - name: relay
            containerPort: 32499
        envFrom:
        - configMapRef:
            name: clusterplex-pms-config
        volumeMounts:
        - name: data
          mountPath: /data
        - name: config
          mountPath: /config
        - name: transcode
          mountPath: /transcode
        resources:          # adapt requests and limits to your needs
          requests:
            cpu: 500m
            memory: 1Gi 
            gpu.intel.com/i915: "1"
          limits:
            gpu.intel.com/i915: "1"
            cpu: 2000m
            memory: 4Gi
      volumes:
      - name: data
        persistentVolumeClaim:
          claimName: "clusterplex-media-pvc"
      - name: config
        persistentVolumeClaim:
          claimName: "clusterplex-config-pvc"
      - name: transcode
        persistentVolumeClaim:
          claimName: "clusterplex-transcode-pvc"
      - name: quicksync
        hostPath: 
          path: /dev/dri
---
apiVersion: v1
kind: Service
metadata:
  name: clusterplex-pms
  labels:
    app.kubernetes.io/name: clusterplex-pms
    app.kubernetes.io/part-of: clusterplex
  namespace: plex
spec:
  type: ClusterIP
  ports:
    - name: pms
      port: 32400
      protocol: TCP
      targetPort: pms
    - name: relay
      port: 32499
      protocol: TCP
      targetPort: relay
  selector:
    app.kubernetes.io/name: clusterplex-pms
    app.kubernetes.io/part-of: clusterplex
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: clusterplex-orchestrator
  labels:
    app.kubernetes.io/name: clusterplex-orchestrator
    app.kubernetes.io/part-of: clusterplex
  namespace: plex
spec:
  replicas: 1
  selector:
    matchLabels:
      app.kubernetes.io/name: clusterplex-orchestrator
      app.kubernetes.io/part-of: clusterplex
  template:
    metadata:
      labels:
        app.kubernetes.io/name: clusterplex-orchestrator
        app.kubernetes.io/part-of: clusterplex
    spec:
      nodeSelector:
        worker: "true"
      containers:
      - name: plex
        image: ghcr.io/pabloromeo/clusterplex_orchestrator:latest
        ports:
          - name: orchestrator
            containerPort: 3500
        envFrom:
        - configMapRef:
            name: clusterplex-orchestrator-config
        resources:          
          requests:
            cpu: 200m
            memory: 64Mi
          limits:
            cpu: 1000m
            memory: 256Mi
---
apiVersion: v1
kind: Service
metadata:
  name: clusterplex-orchestrator
  labels:
    app.kubernetes.io/name: clusterplex-orchestrator
    app.kubernetes.io/part-of: clusterplex
  namespace: plex
spec:
  type: ClusterIP
  ports:
    - name: orchestrator
      port: 3500
      protocol: TCP
      targetPort: orchestrator
  selector:
    app.kubernetes.io/name: clusterplex-orchestrator
    app.kubernetes.io/part-of: clusterplex
---
apiVersion: apps/v1
kind: StatefulSet
metadata:
  name: clusterplex-worker
  labels:
    app.kubernetes.io/name: clusterplex-worker
    app.kubernetes.io/part-of: clusterplex
  namespace: plex
spec:
  serviceName: clusterplex-worker-service
  replicas: 2
  selector:
    matchLabels:
      app.kubernetes.io/name: clusterplex-worker
      app.kubernetes.io/part-of: clusterplex
  template:
    metadata:
      labels:
        app.kubernetes.io/name: clusterplex-worker
        app.kubernetes.io/part-of: clusterplex
    spec:
      affinity:
        podAntiAffinity:
          preferredDuringSchedulingIgnoredDuringExecution:
          - podAffinityTerm:
              labelSelector:
                matchLabels:
                  name: clusterplex-worker
              topologyKey: kubernetes.io/hostname
            weight: 100
          # - podAffinityTerm:
          #     labelSelector:
          #       matchLabels:
          #         name: clusterplex-pms
          #     topologyKey: kubernetes.io/hostname
          #   weight: 50
      containers:
      - name: plex-worker
        image: ghcr.io/pabloromeo/clusterplex_worker:latest   
        ports:
          - name: worker
            containerPort: 3501
        envFrom:
        - configMapRef:
            name: clusterplex-worker-config
        volumeMounts:
        - name: data
          mountPath: /data
        - name: codecs
          mountPath: /codecs
        - name: transcode
          mountPath: /transcode
        - name: config
          mountPath: /config
          readOnly: true
        - name: quicksync 
          mountPath: /dev/dri 

        resources:              # adapt requests and limits to your needs
          requests:
            cpu: 500m
            memory: 1Gi          
            gpu.intel.com/i915: "1"
          limits:
            cpu: 3000m
            memory: 4Gi 
            gpu.intel.com/i915: "1" # requesting 1 GPU
      volumes:
      - name: data
        persistentVolumeClaim:
          claimName: "clusterplex-media-pvc"
      - name: transcode
        persistentVolumeClaim:
          claimName: "clusterplex-transcode-pvc"
      - name: config
        persistentVolumeClaim:
          claimName: "clusterplex-config-pvc"
      - name: quicksync
        hostPath: 
          path: /dev/dri
  volumeClaimTemplates:
    - metadata:
        name: codecs
        labels:
          app.kubernetes.io/name: clusterplex-codecs-pvc
          app.kubernetes.io/part-of: clusterplex
      spec:
        accessModes: [ "ReadWriteOnce" ]
        resources:
          requests:
            storage: 1Gi
        storageClassName: longhorn
```

**Comment by pabloromeo** [comment]
If you switch to the multi-line config for both ipv4 and ipv6, does it work? Meaning:
```
    listen 32499;
    listen [::]:32499;
```

Maybe we can make that the default config and avoid the issue you are seeing.

**Comment by pabloromeo** [comment]
I just released v1.4.13 of Clusterplex, new Helm Chart version 1.1.8 is also available here: https://pabloromeo.github.io/clusterplex/
I've split the listen config of NGINX into the two I mentioned above.
Hopefully that will avoid you having to manually alter the setting.

**Comment by szilardshomelab** [comment]
I tested with the new release and it works perfectly. Thanks!
